lib/job.js

```diff
@@ -191,12 +191,9 @@ Job.prototype = {
 
     this.config.spawn(cmd.command, cmd.args, options, function (err, proc) {
       if (err) {
-        if (err === 127) {
-          return next(127, '', 'Command not found');
-        }
-        
-        console.log('fail!', err, cmd.command, cmd.args);
-        return next(500, '', 'Unable to execude command :( ' + err);
+        // Shell exit code 1 represents general/unknown error
+        var msg = err.message || err.toString() || 'Unknown Error';
+        return next(1, '', 'Unable to execute command: ' + msg);
```

Comment:
I'm really not liking that we use stdout as a way to pass error messages. I feel like we should always return an error object if we have one, and `next(null, exitcode, stdout, stderr)` if not. A "bad" shell code and a js-level error are totally different, no? That would change a semi-public API though.